Scripts/Runtime/Data/VirtualData/VirtualDataIntent.cs

```diff
@@ -0,0 +1,10 @@
+namespace Anvil.Unity.DOTS.Data
+{
+    public enum VirtualDataIntent
+    {
+        //TODO: DISCUSS - Data usage intent
```

Comment:
Data comes in three flavours. 

1. Default - This is data like a TimerInstance or FollowPathInstance where you create it and it exists for as long as there is work to be done on it. It gets updated every frame and then eventually gets Resolved to a result.
2. OneShot - This is like an event. It exists for one frame so someone else can pick it up and spawn whatever they need to. Then this data is gone. No results, just poof.
3. Persistent - This is data that represents State. It exists and maybe gets updated with some new info and looked up when needed but it's not "processed" per frame. It just needs to stay alive. 

I'm not sure about the names. I'm also not sure about using an enum and manually opting into one of these. 
I could make them interfaces instead and you just tag your data so that we can handle the cases. 

`public struct TimerInstance : IKeyedData, IDefaultData`
`public struct WanderStart: IKeyedData, IOneShotData`
`public struct WanderTaskInstance: IKeyedData, IPersistentData`

Thoughts?

Reply:
Interfaces are troublesome because you could stack multiple interfaces on a type and then what would you do?

Are there actual optimizations behind these flavours or is it purely dev convenience?